### PR TITLE
Make things work with apollo 2.0

### PR DESCRIPTION
**Pull request by vincecoppola**
Small changes to make things easier to plug in to apollo 2.0.

### CODE REVIEW

File: src/Cache.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { defaults, get } from 'lodash';
```
Comment (nevir) [inline src/Cache.ts]: nit: let's use the specific `lodash` packages (e.g. `lodash.get`, `lodash.defaults`) to reduce our footprint

File: src/Cache.ts [hunk 2]
```diff
@@ -9,6 +11,12 @@ import { ChangeId, NodeId, Query } from './schema';
 
 export type TransactionCallback = (transaction: CacheTransaction) => void;
 
+export const defaultConfiguration:Configuration = {
+  entityIdForNode(node) {
+    return get(node, 'id', undefined);
```
Comment (nevir) [inline src/Cache.ts]: `undefined` is already the default for this guy

File: src/Cache.ts [hunk 3]
```diff
@@ -17,14 +25,19 @@ export type TransactionCallback = (transaction: CacheTransaction) => void;
...
 export class Cache implements Queryable {
 
-  /** Configuration used by various operations made against the cache. */
-  private readonly _config: Configuration;
-
   /** The current version of the cache. */
-  private _snapshot: CacheSnapshot;
+  private _snapshot: CacheSnapshot = new CacheSnapshot(
+    new GraphSnapshot,
+    new GraphSnapshot,
```
Comment (nevir) [inline src/Cache.ts]: Ah, this should be the same object that's passed in for `baseline` (they should only differ when there are actual optimistic updates active)

File: src/Cache.ts [hunk 4]
```diff
@@ -17,14 +25,19 @@ export type TransactionCallback = (transaction: CacheTransaction) => void;
...
+    new OptimisticUpdateQueue,
+  );
 
   /** All active query observers. */
-  private _observers: QueryObserver[];
+  private _observers: QueryObserver[] = [];
+
+  constructor(private readonly _config: Configuration) {
+    this._config = defaults(_config, defaultConfiguration);
```
Comment (nevir) [inline src/Cache.ts]: 👍

File: src/schema.ts [hunk 5]
```diff
@@ -21,9 +21,9 @@ export type EntityId = NodeId;
  * There are a few pre-defined nodes present in all schemas.
  */
 export enum StaticNodeId {
-  QueryRoot = '☣QueryRoot',
-  MutationRoot = '☣MutationRoot',
-  SubscriptionRoot = '☣SubscriptionRoot',
+  QueryRoot = 'ROOT_QUERY',
```
Comment (nevir) [inline src/schema.ts]: Does Apollo mandate these?

File: src/util/references.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import lodashIsEqual = require('lodash.isequal');
+import lodashIsEqual = require('lodash/isequal');
```
Comment (nevir) [inline src/util/references.ts]: 👎  for reasons above

File: src/Cache.ts (hunk 3 above)
Comment (vincecoppola) [inline src/Cache.ts]: Gotcha, so instantiate it once and then pass the single instance for both, yeah?

File: src/schema.ts (hunk 5 above)
Comment (vincecoppola) [inline src/schema.ts]: It's hard coded in their code, at least. This was what tripped me up for the longest amount of time.

File: src/util/references.ts (hunk 6 above)
Comment (vincecoppola) [inline src/util/references.ts]: This was throwing for me but maybe I messed something up with my packages

File: src/schema.ts (hunk 5 above)
Comment (nevir) [inline src/schema.ts]: 👍

File: src/Cache.ts (hunk 3 above)
Comment (nevir) [inline src/Cache.ts]: Yup!

**Pull request merged**